v3.17/backports/py3-colcon-argcomplete/APKBUILD

```diff
@@ -15,21 +15,30 @@ source="${_pkgname}-${pkgver}.tar.gz::https://github.com/colcon/$_pkgname/archiv
 builddir="$srcdir/$_pkgname-$pkgver"
 
 build() {
-        python3 setup.py build
+        mkdir .venv
+        python3 -m venv .venv
+        . .venv/bin/activate
+        python3 -m pip install -U build
```

Comment:
Please install deps by APKBUILD's dependencies as much as possible (in this case py3-build to makedepends)

Reply:
Default python build tool of the latest abuild seems be gpep517. Does gpep517 work for this?
https://github.com/alpinelinux/abuild/blob/a45d12d4c31413bbbfabd2cb6949e369e78ab806/apkbuild-pypi.in#L31-L42

Reply:
As far as I tried locally, `gpep517` works, but the command `python -m installer ...` causes `FileExistsError: File already exists ...` error. So I think it's better to use `gpep517` and use `pip install` instead of `installer` is fine. sample: https://github.com/seqsense/aports-ros-experimental/pull/928/commits/b216c2a0139d87c32f938cc9d05195a0e7c82dfa

 Do you think this way is OK?

Reply:
`pip install -e .[test]` seems installing the main package as well (https://github.com/pypa/pip/issues/4783) and there are no way to exclude the main package.
In this case, `.testenv/bin/python3 -m pip install .dist/*.whl` outputs `colcon-core is already installed with the same version` and doesn't install the final build version.

Reply:
So, I think
```
.testenv/bin/python3 -m pip install -e .[test]
.testenv/bin/python3 -m pip uninstall -y $_pkgname
.testenv/bin/python3 -m installer .dist/*.whl
```
would be better

Reply:
Done in https://github.com/seqsense/aports-ros-experimental/pull/928/commits/413003a2efb81a1417ba6be1008abdb1f8b7191e